src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_symmetric.cpp

```diff
@@ -0,0 +1,148 @@
+                                               PAL_SymmetricAlgorithm algorithm,
+                                               PAL_ChainingMode chainingMode,
+                                               PAL_PaddingMode paddingMode,
+                                               const uint8_t* pbKey,
+                                               int32_t cbKey,
+                                               const uint8_t* pbIv,
+                                               PAL_SymmetricOptions options,
+                                               CCCryptorRef* ppCryptorOut,
+                                               int32_t* pkCCStatus)
+{
+    if (pbKey == nullptr || cbKey < 1 || ppCryptorOut == nullptr)
+        return -1;
+    if (pbIv == nullptr && chainingMode != PAL_ChainingModeECB)
+        return -1;
+
+// Ensure we aren't passing through things we don't understand
+#if DEBUG
+    switch (operation)
+    {
+        case PAL_OperationEncrypt:
+        case PAL_OperationDecrypt:
+            break;
+        default:
+            return -1;
+    }
+
+    switch (algorithm)
+    {
+        case PAL_AlgorithmAES:
+        case PAL_Algorithm3DES:
+            break;
+        default:
+            return -1;
+    }
+
+    switch (chainingMode)
+    {
+        case PAL_ChainingModeECB:
+        case PAL_ChainingModeCBC:
+            break;
+        default:
+            return -1;
+    }
+
+    switch (paddingMode)
+    {
+        case PAL_PaddingModeNone:
+        case PAL_PaddingModePkcs7:
+            break;
+        default:
+            return -1;
+    }
+
+    if (options != 0)
+        return -1;
+#endif
+
+    CCStatus status = CCCryptorCreateWithMode(operation,
+                                              chainingMode,
+                                              algorithm,
+                                              paddingMode,
+                                              pbIv,
+                                              pbKey,
+                                              static_cast<size_t>(cbKey),
+                                              /* tweak is not supported */ nullptr,
+                                              0,
+                                              /* numRounds is not supported */ 0,
+                                              options,
+                                              ppCryptorOut);
+
+    *pkCCStatus = status;
+    return status == kCCSuccess;
+}
+
+extern "C" int AppleCryptoNative_CryptorUpdate(CCCryptorRef cryptor,
+                                               const uint8_t* pbData,
+                                               int32_t cbData,
+                                               uint32_t* pbOutput,
+                                               int32_t cbOutput,
+                                               int32_t* pcbWritten,
+                                               int32_t* pkCCStatus)
```

Comment:
The only difference (at least for this one) was `pkCCStatus` => `errorCode`; and at the time I felt that they were both the appropriate name for the language.  But that was inconsistent with the rest, and `pkCCStatus` was bad Hungarian anyways, since it's not a pointer to a constant.

So I fixed the C# names to `ccStatus` and the C++ names to `pccStatus`, the "p" getting removed in C# because of the stub -> out translation.